packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -864,13 +872,46 @@ export default class API {
     const resp: { statuses: GitHubCommitStatus[] } = await this.request(
       `${this.originRepoURL}/commits/${sha}/status`,
     );
-    return resp.statuses.map(s => ({
-      context: s.context,
-      // eslint-disable-next-line @typescript-eslint/camelcase
-      target_url: s.target_url,
-      state:
-        s.state === GitHubCommitStatusState.Success ? PreviewState.Success : PreviewState.Other,
-    }));
+    const deploys: GitHubDeployments = await this.request(
+      `${this.originRepoURL}/deployments?task=deploy&ref=${sha}`,
+    );
+    const depStatuses = await Promise.all(
+      deploys.map(async dep => {
+        const stats: GitHubDeployStatuses = await this.request(
+          `${this.originRepoURL}/deployments/${dep.id}/statuses`,
+        );
+
+        return stats.reduce(
+          (acc, stat, i) =>
```

Comment:
Is the purpose of this to return the last successful status or the first one if there is no successful status?

I wonder if something like this would be more clear:
```js
  stats.reverse().forEach(s => {
    if (s.state === GitHubCommitStatusState.Success) {
      return ...
    }
  });

  if (stats.length > 0) {
    return ...
  }
  
  return ...
```

Reply:
In the process of simplifying the type issue, I split the map/reduce, making it much easier to read.